### ISSUE TITLE
Add conversion-option from Bibtex to Mediawiki-reference-style

### ISSUE THREAD

**Issue by lukeflo** [jgm/pandoc#8801]
Hi,

it would be nice if future versions of Pandoc could offer the possibility to convert some **Bibtex-entries** into **Mediawiki-syntax-full-reference**. For example, the input is:

~~~bibtex
@Book{henze_layers_of_time_2000,
  author     = {Henze, Paul B.},
  location   = {London},
  publisher  = {C. Hurst \& Co. Publishers},
  title      = {Layers of Time},
  year       = {2000},
  isbn       = {9781850653936},
  subtitle   = {A History of Ethiopia},
  url        = {https://books.google.de/books?id=gzwoedwOkQMC&pg=PA54&redir_esc=y#v=onepage&q&f=false},
}
~~~
and the output should be ([Wikipedia:Citation Template](https://en.wikipedia.org/wiki/Wikipedia:Citation_templates#)):

~~~mediawiki
{{cite book
| title = Layers of Time: A History of Ethiopia
| last = Henze 
| first = Paul B. 
| year = 2000
| publisher = [[C. Hurst & Co. Publishers]]
| url = https://books.google.com/books?id=gzwoedwOkQMC&pg=PA54
| isbn = 978-1-850-65393-6
}}
~~~

Originally, i posted this question on [StackOverflow](https://stackoverflow.com/questions/76072142/convert-bibtex-entries-to-mediawiki-syntax/76078443), and was told that this option doesn't exist right now. 

For writing a Lua-filter on my own, i'm missing skills and time. But maybe some developers find it appealing to handle this for one of the upcoming releases.

Best

**Comment by erelsgl** [comment]
That would be a great option. The other direction (from MediaWiki citations to BibTeX) would also be very useful. Personally, I am writing both Wikipedia articles and LaTeX papers, and currently I have to write every citations twice, which makes the writing process much slower than it could be.

**Comment by jgm** [comment]
I agree, this would be useful. We already have bidirectional conversion between several bibliographic formats, so it shouldn't be TOO hard to add this...

**Comment by jgm** [comment]
Actually, you might be able to achieve this without any changes in pandoc at all, by using a custom CSL style. (That's an XML file that specifies citation and bibliography formats -- you'd just need to get it to use the format you've given above.)

**Comment by erelsgl** [comment]
Are there instructions on how to build a CSL style?
*--- Erel Segal-Halevi   (http://erelsgl.github.io
<http://erelsgl.github.io>) אראל סגל הלוי*

**Comment by jgm** [comment]
What do you know -- there already is one in the csl styles repository!

Command:
```
pandoc -f biblatex -s -t mediawiki --citeproc --csl https://raw.githubusercontent.com/citation-style-language/styles/1f13ee7e7af91cf6f8bfd0e7ff8aac47dd0e9df2/wikipedia-templates.csl
```

Standard input:
```
@Book{henze_layers_of_time_2000,
  author     = {Henze, Paul B.},
  location   = {London},
  publisher  = {C. Hurst \& Co. Publishers},
  title      = {Layers of Time},
  year       = {2000},
  isbn       = {9781850653936},
  subtitle   = {A History of Ethiopia},
  url        = {https://books.google.de/books?id=gzwoedwOkQMC&pg=PA54&redir_esc=y#v=onepage&q&f=false},
}
```

Output:
```
<div id="refs" class="references csl-bib-body">

<div id="ref-henze_layers_of_time_2000" class="csl-entry">

{{cite book | vauthors=((Henze, P. B.)) | date= 2000 | title=Layers of time: A history of ethiopia | publisher=C. Hurst &amp; Co. Publishers | url=https://books.google.de/books?id=gzwoedwOkQMC&pg=PA54&redir_esc=y#v=onepage&q&f=false | isbn=9781850653936}}


</div>

</div>
```

**Comment by erelsgl** [comment]
@jgm  Thanks! Does it also work in the other direction - for converting from MediaWiki to BibTex?

**Comment by jgm** [comment]
No

**Comment by jgm** [comment]
For the other direction, I think it would make sense to parse the citations into native pandoc citations + references in the metadata. Here's a quick reference to the syntax we'd have to support:
https://en.wikipedia.org/wiki/Help:Citations_quick_reference

**Comment by Dominic-Mayers** [comment]
The solution of [jgm](https://github.com/jgm) has issues. For example, the entry for issn in bibtex can contain two values,  say one for the print version and another for the online version: ` ISSN = {01650106, 15728420}`.  The conversion creates `issn = 01650106, 15728420`, but this parameter only accepts a single value.

**Comment by jgm** [comment]
Fixing that is probably just a matter of modifying the csl style.
